src/element_processing/water_areas.rs

```diff
@@ -36,70 +36,32 @@ pub fn generate_water_areas(editor: &mut WorldEditor, element: &ProcessedRelatio
         }
     }
 
-    // Process each outer polygon individually
-    for (i, outer_nodes) in outers.iter().enumerate() {
```

Comment:
I pretty much fully reverted this. Bug #2 is that processing each polygon individually completely ruins the point of the `merge_loopy_loops` function